### PR TITLE
Update to new Resonance interface and devices

### PR DESCRIPTION
**Pull request by rreiner-hqs**

### CODE REVIEW

File: .cargo/config.toml
Comment (kbarkhqs) [inline .cargo/config.toml:1]: This file should be deleted

File: .gitlab-ci.yml
Comment (kbarkhqs) [inline .gitlab-ci.yml:1]: This should be removed

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,16 @@
 # Changelog
 
-This changelog track changes to the qoqo_iqm project starting at version 0.1.0
+This changelog track changes to the qoqo_iqm project starting at version 0.6.5 (before that it was open-source).
```
Comment (kbarkhqs) [inline CHANGELOG.md:3]:
```suggestion
This changelog track changes to the qoqo_iqm project starting at version 0.1.0.
```

File: deny.toml [hunk 2]
```diff
@@ -151,7 +151,7 @@ ignore = true
 # is only published to private registries, and ignore is true, the crate will
 # not have its license(s) checked
 registries = [
-    #"https://sekretz.com/registry
+    "hqs-ktra"
```
Comment (kbarkhqs) [inline deny.toml:154]: This should be removed

File: documentation/book.toml
Comment (kbarkhqs) [inline documentation/book.toml:1]: I'm not sure if these files should be present, I forget what our documentation set-up is for github

File: documentation/src/Introduction.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Internal qoqo-iqm User Guide
```
[the remark this answers is not in the document]
Comment (kbarkhqs) [reply documentation/src/Introduction.md:1]: Only change this if we keep it

File: documentation/src/SUMMARY.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Internal qoqo-iqm User Guide
```
Comment (kbarkhqs) [inline documentation/src/SUMMARY.md:1]:
```suggestion
# qoqo-iqm User Guide
```

Comment (kbarkhqs) [reply documentation/src/SUMMARY.md:1]: Same here

File: qoqo-iqm/Cargo.toml [hunk 5]
```diff
@@ -1,32 +1,33 @@
...
-readme = "README.md"
-edition = "2021"
-rust-version = "1.56"
+readme = "../README.md"
 categories = ["science", "simulation"]
 description = "IQM backend for qoqo quantum computing toolkit"
-include = ["src*", "qoqo_iqm", "LICENSE", "pyproject.toml", "README.md"]
-
+include = ["src", "qoqo_iqm", "LICENSE", "pyproject.toml", "Cargo.toml"]
+publish = ["hqs-ktra"]
```
Comment (kbarkhqs) [inline qoqo-iqm/Cargo.toml:15]:
```suggestion
```

File: qoqo-iqm/Cargo.toml [hunk 6]
```diff
@@ -1,32 +1,33 @@
...
+doctest = false
 
 [dependencies.pyo3]
-version = "0.18"
+version = "0.20"
 features = ["num-complex"]
 
 [dependencies]
-roqoqo-iqm = {version="0.6", path="../roqoqo-iqm"}
+roqoqo-iqm = {version="0.7", path="../roqoqo-iqm", registry="hqs-ktra"}
```
Comment (kbarkhqs) [inline qoqo-iqm/Cargo.toml:29]:
```suggestion
roqoqo-iqm = {version="0.7", path="../roqoqo-iqm"}
```

File: qoqo-iqm/docs/index.rst
Comment (kbarkhqs) [inline qoqo-iqm/docs/index.rst:1]: This content should be reverted

File: documentation/book.toml
Comment (rreiner-hqs) [reply documentation/book.toml:1]: I have removed the whole documentation directory

**Review by kbarkhqs — APPROVED** [review]

**Pull request merged**